### ISSUE TITLE
StaleBot should not close defects

### ISSUE THREAD

**Issue by GregorKopka** [openzfs/zfs#10793]
### System information
GitHub

### Describe the problem you're observing
I see a bot marking issues as stale regardless of other labels.
Issues about a **defect** should stay open until the defect has been resolved, this bot thinks otherwise.

### Describe how to reproduce the problem
https://github.com/openzfs/zfs/issues?q=is%3Aissue+is%3Aopen+label%3A%22Status%3A+Stale%22++label%3A%22Type%3A+Defect%22++

**Comment by behlendorf** [comment]
The bot won't tag any issue as stale or close it if it's tagged with `Type: Feature` or `Status: Understood`.  We could consider adding `Type: Defect` to the list, the concern is that we are now automatically tagging new bug reports with that label so they would all be excluded.  We'd want to come up with a tag for new issues which still need to be triaged to determine if they're defects or not.

**Comment by zenaan** [comment]
See also tag "good first issue" and e.g. here:

https://github.com/openzfs/zfs/issues/5229#issuecomment-679423419

**Comment by zenaan** [comment]
Also tags "RFC" and "For discussion".

**Comment by zenaan** [comment]
On Tue, Aug 25, 2020 at 01:00:20PM +1000, Zenaan Harkness wrote:
> Also tags "RFC" and "For discussion".


Sorry "RFC" was in the subject, the tag is "Question" see e.g. these 2 examples:

https://github.com/openzfs/zfs/issues/3492#issuecomment-679440880

https://github.com/openzfs/zfs/issues/3436#issuecomment-679440886

**Comment by zenaan** [comment]
Also "Status: Design Review Needed", e.g.:

https://github.com/openzfs/zfs/issues/3049#issuecomment-679440928

**Comment by bunder2015** [comment]
How do I mute these, my inbox has been spammed hard in the past 24h.

**Comment by gmelikov** [comment]
@bunder2015 it's a one time storm to work through all issues, you may just mark them read or delete by sender (`stale[bot]`).

**Comment by Ornias1993** [comment]
@GregorKopka @zenaan
To explain a bit of thought behind the bot:
- Many older issues are not verified to be currently even existing. If you can verify them to be actually still present, you can just override the bot. 
- Excluding "Type: Defect" means totally removing the bot, it's designed especially with defects in mind
- Please be aware the bot only requires you to overrule it once a year and not more. If you really believe the issue is still relevant, it's a rather small act to just comment on the bot to remove the flag.
- "Good first issue", "Status: Design Review Needed" etc. do not mean something is still a relevant issue. (see previous
- Simply put: The idea is to force submitters of issues (and commenters) to make sure issues are still relevant. We have issues that are 7(!) years old and no one knows if they are still relevant.
- Also: Abusing a the system by flagging your question as a defect (it's clearly not a bug, if you read the PR and the issue about it), is not good form

@bunder2015
- Sorry for the inconvenience, this is only an issue the first 3 days when it processes the backlog

@behlendorf This isn't a bug, can be reclassivied as "Type: Question"

**Comment by Ornias1993** [comment]
TLDR:
You as submitter are responsible for making sure your issue is still relevant after a year.
It's not a big thing to ask IMHO.

OpenZFS doesn't have a staff on hand to daily go test all sorts of old issues for validity. It's already hard enough to even make sure everything is tagged correctly.

**Comment by drescherjm** [comment]
> OpenZFS doesn't have a staff on hand to daily go test all sorts of old issues for validity. It's already hard enough to even make sure everything is tagged correctly.

I think this is something the community could help with if a question has detailed instructions how to reproduce. However I do realize that most of these old bug reports will not have that.

**Comment by Ornias1993** [comment]
@drescherjm Exactly! 
Thats the thing the bot is for: reminding people to confirm issues and if no one responds anything of value in months it gets closed (with the option to reopen at a later date)

To be clear:  It doesn't have to be the original opener of the Issue that responds to stop the Bot from closing.

**Comment by Ornias1993** [comment]
I want to throw in a example of what we all hope to see  happening more with this bot:
https://github.com/openzfs/zfs/issues/6671

Closing ourselves really is a last resort, but there already are multiple issue creators who removed their issue after being reminded it's quite old. 👍

**Comment by Ornias1993** [comment]
@zenaan I've found the time to review the issues you linked...
Lets go over them:

> #3492 (comment) 

- I don't see why this is relevant, it's a forward from FreeBSD (which this codebase by now moved quite far away from by now) from 2015 with no actual bug or error being known. It's flagged "question", while there isn't really any question and there also isn't likely to be any response anytime soon. This is a good example of an issue that shouldn've been closed or forwarded to the mailinglist (and possibly left there) quite some time ago.
It's most likely flagged question because someone back in 2015 didn't really know what the heck to do with this, which i don't blame the guy/gal

#3436 (comment)

- This one is actually a question "Comments ?", which is good. And the answer the community has given is quite clear: "No.".
When it comes to the content of the question (a forwarded issue from FreeBSD codebase) Same issue as #3492 applies: It might not ever have been an issue in the first place and the code bases have diverged even more.
Actual questions should actually be closed sooner rather than later imho, because if they aren't answered in the first months they are almost never answered anyway (they aren't really supposed to be asked here in the first place, but should primarily be put on the mailinglist, it's just being condoned)

#3049 (comment)

- Tagging here seems to be another case of "I'll just tag this with something because I don't really know what to do with this". It's a good question, which gets answered. But it does certainly not contain any design reviews.
As it has been answered and the original submitter has not made any attempt to ask further questions.

I think all these cases highlight why we would want something to garbage clean once in a while. While they certainly wheren't garbage when they where made, they currently add nothing to the issuelist today and it's doubtfull they will get any use anytime soon.  If the original submitter or someone else disagrees they can always halt the bot closing the issue in 3 months.
